mlflow/deployments/databricks/__init__.py

```diff
@@ -351,42 +352,67 @@ def create_endpoint(self, name, config=None, route_optimized=False):
 
             client = get_deploy_client("databricks")
             endpoint = client.create_endpoint(
-                name="chat",
                 config={
-                    "served_entities": [
-                        {
-                            "name": "test",
-                            "external_model": {
-                                "name": "gpt-4",
-                                "provider": "openai",
-                                "task": "llm/v1/chat",
-                                "openai_config": {
-                                    "openai_api_key": "{{secrets/scope/key}}",
+                    "name": "test",
+                    "config": {
+                        "served_entities": [
+                            {
+                                "external_model": {
+                                    "name": "gpt-4",
+                                    "provider": "openai",
+                                    "task": "llm/v1/chat",
+                                    "openai_config": {
+                                        "openai_api_key": "{{secrets/scope/key}}",
+                                    },
                                 },
-                            },
-                        }
-                    ],
+                            }
+                        ],
+                        "route_optimized": True,
+                    },
                 },
-                route_optimized=True,
             )
             assert endpoint == {
-                "name": "chat",
+                "name": "test",
                 "creator": "alice@company.com",
                 "creation_timestamp": 0,
                 "last_updated_timestamp": 0,
                 "state": {...},
                 "config": {...},
                 "tags": [...],
                 "id": "88fd3f75a0d24b0380ddc40484d7a31b",
+                "permission_level": "CAN_MANAGE",
+                "route_optimized": False,
+                "task": "llm/v1/chat",
+                "endpoint_type": "EXTERNAL_MODEL",
+                "creator_display_name": "Alice",
+                "creator_kind": "User",
             }
 
         """
-        config = config.copy() if config else {}  # avoid mutating config
-        extras = {}
-        for key in ("tags", "rate_limits"):
-            if tags := config.pop(key, None):
-                extras[key] = tags
-        payload = {"name": name, "config": config, "route_optimized": route_optimized, **extras}
+        if config and "config" in config:
+            # config contains the full API request payload
+            if route_optimized or name:
```

Comment:
Is this safe? 

What if I define `name` at the top level arg and don't put it in the `config` kwargs? That would be the old way of defining the endpoint and we shouldn't break that experience. 

We can update this to:
1. Check if `name` is defined in `config`. 
a. If it is, check that name matches the arg `name`. 
    I. If they match, issue a deprecation warning about the usage of the top-level `name` argument. 
    II. If they do not match, raise an MLflowException explaining that both cannot be defined and that `name` should be defined within config, messaging that the `name` named arg is deprecated. 
b. If it is not, then ensure that `name` is set (it is a required arg that is now synthetically made optional) and raise an MLflowException if it is not set, indicating to users to supply it within `config`. 

For the `route_optimized` argument, similar changes will be needed, but due to the default value that used to exist, the logic will need to be slightly different.

Reply:
I think the earlier version was safe and enforced a stricter separation between the "old" and "new" approaches—it would, for example, error if a user put name as a param and in the config, even if they were the same.

I updated according to your suggestions—I think the approach of checking if they are the same and issuing a warning, and only erroring if there is a name mismatch, is more forgiving.